File: scripts/generate-icons.py
import cairosvg

cairosvg.svg2png(url='public/icons/fable-leaf.svg', write_to='public/icons/icon-512.png', output_width=512, output_height=512)
print('✓ icon-512.png')

cairosvg.svg2png(url='public/icons/fable-leaf.svg', write_to='public/icons/icon-192.png', output_width=192, output_height=192)
print('✓ icon-192.png')

cairosvg.svg2png(url='public/icons/fable-leaf.svg', write_to='public/apple-icon.png', output_width=180, output_height=180)
print('✓ apple-icon.png')

print('Done.')

File: scripts/generate-icons.mjs
import sharp from 'sharp'
import { readFileSync } from 'fs'
import { join, dirname } from 'path'
import { fileURLToPath } from 'url'

const __dirname = dirname(fileURLToPath(import.meta.url))
const root = join(__dirname, '..')

const svg = readFileSync(join(root, 'public/icons/fable-leaf.svg'))

await sharp(svg).resize(512, 512).png().toFile(join(root, 'public/icons/icon-512.png'))
console.log('✓ icon-512.png')

await sharp(svg).resize(192, 192).png().toFile(join(root, 'public/icons/icon-192.png'))
console.log('✓ icon-192.png')

await sharp(svg).resize(180, 180).png().toFile(join(root, 'public/apple-icon.png'))
console.log('✓ apple-icon.png')

console.log('Done.')
